### PR TITLE
[Fix] Use delta of Root class permissions

### PR DESCRIPTION
**Pull request by shpran**
# [ADF-1541](https://oat-sa.atlassian.net/browse/ADF-1541)

## Changelog
* Instead of calculating permissions delta for every class, we will use permissions delta of the root class
* Added new task parameter to provide permissions delta
* Added new command parameter to provide permissions delta

[ADF-1541]: https://oat-sa.atlassian.net/browse/ADF-1541?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: model/tasks/ChangePermissionsTask.php [hunk 1]
```diff
@@ -84,16 +86,20 @@ private function doHandle(
         string $requestRoot,
         array $privileges,
         bool $isRecursive,
-        bool $withNestedResources
+        bool $withNestedResources,
+        ?array $permissionsDelta
     ): Report {
+        $permissionService = $this->getPermissionService();
+        $permissionsDelta ??= $permissionService->getPermissionsDelta($root, $privileges);
```
Comment (gabrielfs7) [inline model/tasks/ChangePermissionsTask.php:93]: Just to be sure, this will work on PHP 7.4, correct?

**Comment by hectoras** [comment]
`ChangePermissionsCommandTest` is now failing, could you please take a look at? We now need to pass an additional parameter to `ChangePermissionsCommand`, probably passing an empty array would suffice to fix the current tests :slightly_smiling_face:

File: model/tasks/ChangePermissionsTask.php (hunk 1 above)
Comment (shpran) [reply model/tasks/ChangePermissionsTask.php:93]: Yes: https://www.php.net/manual/en/migration74.new-features.php#migration74.new-features.core.null-coalescing-assignment-operator

**Comment by shpran** [comment]
> `ChangePermissionsCommandTest` is now failing, could you please take a look at? We now need to pass an additional parameter to `ChangePermissionsCommand`, probably passing an empty array would suffice to fix the current tests 🙂

Already fixed, thanks 🙂

File: model/tasks/ChangePermissionsTask.php (hunk 1 above)
Comment (hectoras) [reply model/tasks/ChangePermissionsTask.php:93]: I think it will: https://wiki.php.net/rfc/null_coalesce_equal_operator

**Review by hectoras — APPROVED** [review]
- [x] Pull request title and description are meaningful
- [x] Pull request's target is not `master`
- [x] Commits are following conventional commits
- [x] New code is covered by tests (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [x] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [x] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Kitchen links are not exposed in the description
- [x] No unnecessary dependencies are installed
- [x] Target branch is correct
- [x] Change log provided

**Review by gabrielfs7 — APPROVED** [review]
Amazing job @shpran 🎉 

- [x] New code is covered by tests (if applicable)
- [x] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [x] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [ ] Acceptance criteria are respected
- [ ] Pull request title and description are meaningful

**Pull request merged**